onesixtyone.c

```diff
@@ -121,11 +131,71 @@ void read_communities(char* filename)
 			exit(1);
 		}
 	}
-			
+
 	community_count = i;
 	fclose(fd);
 }
 
+int add_host(const char *ipmask)
+{
+  int ret = -1;
+  char *addr = NULL;
+  char *slash;
+  struct addrinfo hints;
+  struct addrinfo *result = NULL;
+  unsigned long longtmp, i, ips;
+  struct in_addr startaddr;
+  struct in_addr endaddr;
+  int netmask;
+
+  addr = strdup(ipmask);
+  slash = strchr(addr, '/');
+  if (slash == NULL) {
+    netmask = 32;
+  } else {
+    netmask = atoi(slash + 1);
+    if (netmask <= 0 || netmask > 32) {
+      goto OUT;
+    }
+    *slash = '\0';
+  }
+
+
+  memset(&hints, 0, sizeof(struct addrinfo));
+  hints.ai_family = AF_INET;	 /* Allow IPv4 */
+  hints.ai_socktype = SOCK_DGRAM; /* Datagram socket */
+  hints.ai_flags =  AI_NUMERICHOST|AI_NUMERICSERV; /* no dns */
+  hints.ai_protocol = 0;			 /* IPv4 */
+  hints.ai_canonname = NULL;
+  hints.ai_addr = NULL;
+  hints.ai_next = NULL;
+
+  if (getaddrinfo(addr, NULL, &hints, &result) != 0) {
+    goto OUT;
+  }
+
+  longtmp = ntohl(((struct sockaddr_in *)result->ai_addr)->sin_addr.s_addr);
+  startaddr.s_addr = longtmp & (unsigned long) (0 - (1<<(32 - netmask)));
```

Comment:
let's change longtmp to `in_addr_t` and remove the `(unsigned long)` casts on these 2 lines